### PR TITLE
refactor singledocumentbuilder

### PR DESCRIPTION
**Pull request by ansyral**
1. move the methods to load hostservice to interface IHostServiceConstructor
2. move the methods to prebuild/build/postbuild to interface IPhaseHandler

@vwxyzh @chenkennt @superyyrrzz @hellosnow @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructorWithIncremental.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Build.Engine.Incrementals;
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    internal class HostServiceConstructorWithIncremental : HostServiceConstructor, IHostServiceConstructor
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructorWithIncremental.cs]: >IHostServiceConstructor [](start = 83, length = 23)

remove

File: src/Microsoft.DocAsCode.Build.Engine/PostbuildPhaseHandlerWithIncremental.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Build.Engine.Incrementals;
+    using Microsoft.DocAsCode.Plugins;
+
+    internal class PostbuildPhaseHandlerWithIncremental : PostbuildPhaseHandler, IPhaseHandler
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/PostbuildPhaseHandlerWithIncremental.cs]: >IPhaseHandler [](start = 81, length = 13)

remove

File: src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructor.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.IO;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    internal class HostServiceConstructor : IHostServiceConstructor
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructor.cs]: >Constructor [](start = 30, length = 11)

rename to Creator?

**Comment by vwxyzh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructor.cs (hunk 3 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructor.cs]: ha, any difference?

---
In reply to: [89049826](https://github.com/dotnet/docfx/pull/913#discussion_r89049826) [](ancestors = 89049826)

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/HostServiceConstructor.cs]: like NaveVisitorCreator, CommandCreator... Just minor :)

---
In reply to: [89050964](https://github.com/dotnet/docfx/pull/913#discussion_r89050964) [](ancestors = 89050964,89049826)

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**